### PR TITLE
Reconnection logic

### PR DESCRIPTION
**Pull request by nytamin**
This PR adds several major features, as well as some minor fixes.

(This is a continuation of #10 )

## BREAKING: Yeelight.sendCommand now rejects promise if response from bulb is an error
Before, the sendCommand method always resolved the promise, no matter if the response from the bulb was an affirmative or an error. I've changed this so it rejects the promise if the bulb returns an error.

This is a breaking change, because it affects the interface to the consumer of the library.

## Infinite discovery
I wanted to be able to set the `Discover` class to continously sweep the network after devices. I reworked the logics in `Discover.start()` so that the promise won't resolve/reject but instead continue to sweep infinitely (or until `Discover.destroy()`
To do infinite discovery:
```
const discover = new Discover({
    limit: 0,
    timeout: 0
});
discover.on("deviceAdded", (device) => {
})
discover.start()
```
## Connection status & ping
I've added logic so that the Yeelight class emits "connected" & "disconnected" statuses.
Keeping track of the connection status by pinging the bulb regularly.

## Automatic reconnection
I've added logics for detecting when a connection has closed/failed and methods for reconnecting.
To activate:
```
const yeelight = new Yeelight({
    lightIp: '127.0.0.1'
});
yeelight.autoReconnect = true
yeelight.on('connected', () => {
    // This will change the light color every time the light is reconnected
    yeelight.setBright(1)
    .catch(e => console.log('error in setBright', e))
    yeelight.setRGB(new Color(
        Math.round(Math.random()* 255),
        Math.round(Math.random()* 255),
        Math.round(Math.random()* 255)
    ), 'sudden')
    .catch(e => console.log('error in setRGB', e))
})
yeelight.connect()
```

### CODE REVIEW

**Comment by nytamin** [comment]
More features:
## Connect to bulb using Id only
Added convenience function, so it's possible to connect to a bulb, using the ID of the bulb
Example:
```
const yeelight = new Yeelight({
    lightId: '0x0000000007c9505f' // whoa, by id only? amaaazing!
});
yeelight.autoReconnect = true
yeelight.on('connected', () => {
    // This will change the light color every time the light is reconnected
    yeelight.setRGB(new Color(
        Math.round(Math.random()* 255),
        Math.round(Math.random()* 255),
        Math.round(Math.random()* 255)
    ), 'sudden')
    .catch(e => console.log('error in setRGB', e))
})
yeelight.connect()
.catch(e => console.log('error in connect', e))
```

**Comment by samuraitruong** [comment]
Lets me know when you happy to merge this. Then we can release a new version. Make sure you update index.d.ts. Thanks

File: src/models/color.ts [hunk 1]
```diff
@@ -11,6 +11,6 @@ export class Color {
     }
     public getValue() {
         // tslint:disable-next-line:no-bitwise
-        return this.red * 65535 + this.green * 256 + this.blue;
+        return this.red * 0x10000 + this.green * 0x100 + this.blue;
```
Comment (samuraitruong) [inline src/models/color.ts]: I prefer not using bitwise, just correct the number above

Comment (nytamin) [reply src/models/color.ts]: fair enough, fixed! :)

**Comment by nytamin** [comment]
The placement of index.d.ts looked a bit off, usually it's located together with the other generated js-files in /lib. I removed it and instead added an option in tsconfig.json, so it will be generated automatically when building.

I've also added an .npmignore, so only the build artifacts are published to npm (the source-files should not be included).

I'm happy with this now, feel free to merge it :)

**Pull request merged**